generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -0,0 +1,180 @@
+package com.scottlogic.deg.generator.constraints;
+
+import com.scottlogic.deg.generator.constraints.atomic.*;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StringConstraintsCollection {
```

Comment:
From what i see, the way this class is meant to be used is 
intersect
isContradictory

so, merge, then check if the merge contradicts.

This class could do with changes to reflect this

Reply:
This type is trying to reflect data, with the ability to operate on the data.
As such it can then be used in the merger rather than being the merger itself.

Reply:
Agreed to review the 'merging' ecosystem as a separate task to follow this PR